src/schema/resolvers/invitation/mutation.resolver.js

```diff
@@ -0,0 +1,122 @@
+import { invitationTriggers } from "utils/pubSubTriggers";
+import { fromInvitationDtoToInvitation } from "utils/mapper/invitation";
+
+const createInvitation = async (_, args, context) => {
+  // get Event for invitation
+  const { data: eventData } = await context.database.events.getById(
+    args.eventId
+  );
+
+  // calculate random 6 digit generated code
+  const validCode = false;
+  const { data: invitations } = await context.database.invitations.find({
+    eventId: args.eventId,
+  });
+  let randomCode = "";
+  while (!validCode) {
+    const code = Math.floor(100000 + Math.random() * 900000).toString();
+    const checkExistingCode = invitations.filter(
+      (invitation) => invitation.code === code
+    );
+    if (checkExistingCode.length === 0) {
+      randomCode = code;
+      break;
+    }
+  }
+
+  const currentDateTime = new Date().toLocaleString();
+
+  //  create invitation object
+  const invitation = {
+    eventId: args.eventId,
+    attendeeId: args.userId,
+    code: randomCode,
+    numAttendees: args.numAttendees,
+    status: eventData.autoAcccept ? "ACCEPTED" : "PENDING",
+    claimedDate: currentDateTime,
+  };
+
+  // insert invitation
+  const { data: invitationData, statusCode } =
+    await context.database.invitations.create(invitation);
+
+  if (statusCode === 200) {
+    const newInvitation = fromInvitationDtoToInvitation(
```

Comment:
Don't need to do this at all.
Make an invitation.resolver.js something like

```
const event = async (parent, _, context) => {
  const { data } = await context.database.events.find({
    id: parent.eventId,
  });
  return data;
};

const Invitation = {
  event,
};

export default {
  Invitation,
};
```

Reply:
ah i'm dumb! this makes way more sense thank you